pkg/util/runtime.go

```diff
@@ -45,21 +45,17 @@ func GetClientReaderFromClient(c client.Client) (client.Reader, error) {
 	if dr, err := getDelegatingReader(c); err != nil {
 		return nil, err
 	} else {
-		return dr.ClientReader, nil
+		return dr, nil
 	}
 }
 
 // getDelegatingReader try to extract DelegatingReader from client.
-func getDelegatingReader(c client.Client) (*client.DelegatingReader, error) {
-	dc, ok := c.(*client.DelegatingClient)
+func getDelegatingReader(c client.Client) (client.Reader, error) {
```

Comment:
@SimonCqk check this change, is the  DelegatingReader removed ? 
So it's always getting from the cache, you previous patch is useless in this case.

this method can also be merged with GetClientReaderFromClient

Reply:
solved by using controller-runtime v0.6.2 (can work with k8s v20.7